usageReports/ec2/prepare_response.go

```diff
@@ -99,15 +101,20 @@ func addCostToReport(report ec2.Report, costs ResponseCost) (ec2.Report) {
 
 // prepareResponseEc2 parses the results from elasticsearch and returns the EC2 usage report
 func prepareResponseEc2(ctx context.Context, resEc2 *elastic.SearchResult, resCost *elastic.SearchResult) ([]ec2.Report, error) {
-	parsedEc2  := ResponseEc2{}
+	logger := jsonlog.LoggerFromContextOrDefault(ctx)
+	parsedEc2 := ResponseEc2{}
 	parsedCost := ResponseCost{}
 	reports := []ec2.Report{}
 	err := json.Unmarshal(*resEc2.Aggregations["top_reports"], &parsedEc2.TopReports)
 	if err != nil {
+		logger.Error("Error while unmarshaling ES EC2 response", err)
 		return nil, err
 	}
 	if resCost != nil {
-		json.Unmarshal(*resCost.Aggregations["accounts"], &parsedCost.Accounts)
+		err = json.Unmarshal(*resCost.Aggregations["accounts"], &parsedCost.Accounts)
+		if err != nil {
+			logger.Error("Error while unmarshaling ES cost response", err)
```

Comment:
If there is an error unmarshalling data, shouldn't you return ? Since your `for` loop after that will do nothing anyway